packages/fxa-auth-server/lib/error.js

```diff
@@ -1418,7 +1418,7 @@ AppError.iapInvalidToken = (error) => {
       code: 400,
       error: 'Bad Request',
       errno: ERRNO.IAP_INVALID_TOKEN,
-      message: 'Invalid token',
+      message: `Invalid IAP token${error?.message ? `: ${error.message}` : ''}`,
```

Comment:
@bbangert : I am trying to implement a minimal fix to help the VPN team understand why some existing Apple IAP users might have failed to migrate if we received an App Store Server API error.

Rather than create new types for [all possible `SubscriptionStatuses` API errors](https://developer.apple.com/documentation/appstoreserverapi/get_all_subscription_statuses) and have more complex [library error wrapping](https://github.com/mozilla/fxa/blob/605743f52a2af9eed901a2b43a0103b5fb03d3ff/packages/fxa-shared/payments/iap/apple-app-store/purchase-manager.ts#L257-L271) logic, I opted to pass along whatever error message we get. Does this seem acceptable to you? If not, what would you recommend?

Reply:
Does the message here carry over the details needed? I recall we had the other place where we masked the underlying Apple error, is that retained now?

Reply:
No we mask the underlying Apple error -- all the caller sees right now, regardless of the Apple API error is 400 Bad Request "Invalid token".

Reply:
I mean, what are the possible contents of the `error.message` here?

Reply:
The error messages for [all of the possible 4xx and 5xx errors from the `SubscriptionStatuses` API](https://developer.apple.com/documentation/appstoreserverapi/get_all_subscription_statuses). E.g. `"Invalid request app identifier."` for https://developer.apple.com/documentation/appstoreserverapi/invalidappidentifiererror.